Description: FHS paths for nodejs
 Use /usr/lib/nodejs for packaged modules.
 Add /usr/local/lib/nodejs for user-installed modules.
 Use /usr/include/nodejs for headers.
 
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2011-02-13
--- a/src/node_config.h.in
+++ b/src/node_config.h.in
@@ -22,7 +22,7 @@
 #ifndef NODE_CONFIG_H
 #define NODE_CONFIG_H
 
-#define NODE_CFLAGS "@CCFLAGS@ @CPPFLAGS@ -I@PREFIX@/include/node"
+#define NODE_CFLAGS "@CCFLAGS@ @CPPFLAGS@ -I@PREFIX@/include/nodejs"
 #define NODE_PREFIX "@PREFIX@"
 
 #endif /* NODE_CONFIG_H */
--- a/tools/wafadmin/Tools/node_addon.py
+++ b/tools/wafadmin/Tools/node_addon.py
@@ -28,7 +28,7 @@
   lib = join(prefix, 'lib')
 
   conf.env['LIBPATH_NODE'] = lib
-  conf.env['CPPPATH_NODE'] = join(prefix, 'include', 'node')
+  conf.env['CPPPATH_NODE'] = join(prefix, 'include', 'nodejs')
 
   conf.env.append_value('CPPFLAGS_NODE', '-D_GNU_SOURCE')
   conf.env.append_value('CPPFLAGS_NODE', '-DEV_MULTIPLICITY=0')
@@ -64,7 +64,7 @@
     nodePath = None
     if not os.environ.has_key('NODE_PATH'):
         if not os.environ.has_key('HOME'):
-            nodePath = join(get_prefix(), 'lib', 'node')
+            nodePath = join(get_prefix(), 'lib', 'nodejs')
         else:
             nodePath = join(os.environ['HOME'], '.node_libraries')
     else:
--- a/tools/node-waf
+++ b/tools/node-waf
@@ -3,9 +3,11 @@
 
 
 join = os.path.join
-bindir = os.path.dirname(os.path.realpath(__file__))
-prefix = join(bindir, "..")
-wafdir = join(prefix, "lib", "node")
+if os.environ.has_key('PREFIX_NODE'):
+	prefix = os.environ['PREFIX_NODE']
+else:
+	prefix = "/usr"
+wafdir = join(prefix, "share", "nodejs")
 
 w = join(wafdir, 'wafadmin')
 t = join(w, 'Tools')
--- a/lib/module.js
+++ b/lib/module.js
@@ -422,7 +422,7 @@
 };
 
 Module._initPaths = function() {
-  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node')];
+  var paths = ['/usr/local/lib/nodejs', '/usr/lib/nodejs'];
 
   if (process.env['HOME']) {
     paths.unshift(path.resolve(process.env['HOME'], '.node_libraries'));
--- a/tools/nodejs.pc.in
+++ b/tools/nodejs.pc.in
@@ -1,7 +1,7 @@
 version=@VERSION@
 prefix=@PREFIX@
 libdir=${prefix}/lib
-includedir=${prefix}/include/node
+includedir=${prefix}/include/nodejs
 
 Name: nodejs
 Description: Evented I/O for V8 JavaScript.
